documentation/src/main/asciidoc/topics/con_off_heap_memory.adoc

```diff
@@ -0,0 +1,61 @@
+[id='off_heap_memory']
+= Off Heap Memory
+{brandname} can store cache entries in JVM heap memory or off heap.
+Off heap storage lets your Java objects occupy native memory outside the managed JVM memory space.
+
+Off heap storage requires less overhead per entry than in JVM heap memory.
+In other words, off heap storage uses less memory than heap storage for the same amount of data.
+Another benefit of off heap storage is that it is not affected by Garbage Collector runs, which can improve overall JVM performance.
+There are some trade-offs with off heap storage; for example, JVM heap dumps do not show entries stored in off heap memory.
+
+Consider the following simplified illustration of memory space for a JVM process where {brandname} is running:
+
+image::offheap.svg[]
+
+.JVM heap memory
+The heap is divided into young and old generations that help keep referenced Java objects and other application data in memory.
+The Garbage Collector (GC) process reclaims space from unreachable objects, running more frequently on the young generation memory pool.
+
+When {brandname} stores cache entries in JVM heap memory, GC runs can take longer to complete as you start adding data to your caches.
+Because GC is a fairly intensive process, longer and more frequent runs can degrade application performance.
+
+.Off heap memory
+Off heap memory is native available system memory outside JVM memory management.
+The preceding diagram shows the `Metaspace` memory pool that holds class metadata and is allocated from native memory.
+The diagram also represents a section of native memory that holds {brandname} cache entries.
+
+.Storing data off heap
+When you add entries to off heap caches, {brandname} dynamically allocates native memory to your data.
+
+{brandname} hashes the serialized `byte[]` for each key into buckets that are similar to a standard Java `HashMap`.
```

Comment:
We should probably mention that this means that equality is done via the serialized bytes and not the object representation as well. I believe we had that somewhere before... but don't remember where.

Reply:
@wburns Please check the note I added about equality: https://github.com/infinispan/infinispan/pull/8803/files#diff-b632766944b1fc30eb18b540ff43965eb166f1f10211f64cab65b295daa5a70eR39